From 8f2b5fd545ef65c12e12067c7edd6a084240c380 Mon Sep 17 00:00:00 2001
From: "Du, Changbin" <changbinx.du@intel.com>
Date: Wed, 5 Nov 2014 12:15:08 +0800
Subject: [PATCH 011/441] usb: phy-fsm: fix state transition from A_WAIT_VRISE
 to next

Per OTG Spec, the correct transition is:
A_WAIT_VRISE -> A_WAIT_VFALL if id/a_bus_drop/a_wait_vrise_tmout asserted
A_WAIT_VRISE -> A_WAIT_BCON if a_vbus_vld asserted

Change-Id: I9337ac0dbf779c37c3491126bd0919ab6986b9c1
Signed-off-by: Du, Changbin <changbinx.du@intel.com>
---
 drivers/usb/phy/phy-fsm-usb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/usb/phy/phy-fsm-usb.c
+++ b/drivers/usb/phy/phy-fsm-usb.c
@@ -287,10 +287,10 @@ int otg_statemachine(struct otg_fsm *fsm
 			otg_set_state(fsm, OTG_STATE_A_WAIT_VRISE);
 		break;
 	case OTG_STATE_A_WAIT_VRISE:
-		if (fsm->id || fsm->a_bus_drop || fsm->a_vbus_vld ||
-				fsm->a_wait_vrise_tmout) {
+		if (fsm->id || fsm->a_bus_drop || fsm->a_wait_vrise_tmout)
+			otg_set_state(fsm, OTG_STATE_A_WAIT_VFALL);
+		else if (fsm->a_vbus_vld)
 			otg_set_state(fsm, OTG_STATE_A_WAIT_BCON);
-		}
 		break;
 	case OTG_STATE_A_WAIT_BCON:
 		if (!fsm->a_vbus_vld)
